services/m3dbnode/config/fs.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package config
+
+import (
+	"fmt"
+	"os"
+)
+
+const (
+	// DefaultNewFileMode is the default new file mode.
+	DefaultNewFileMode = os.FileMode(0666)
+	// DefaultNewDirectoryMode is the default new directory mode.
+	DefaultNewDirectoryMode = os.FileMode(0755)
+)
+
+// FilesystemConfiguration is the filesystem configuration.
+type FilesystemConfiguration struct {
+	// File path prefix for reading/writing TSDB files
+	FilePathPrefix string `yaml:"filePathPrefix" validate:"nonzero"`
+
+	// Write buffer size
+	WriteBufferSize int `yaml:"writeBufferSize" validate:"min=0"`
+
+	// Read buffer size
+	ReadBufferSize int `yaml:"readBufferSize" validate:"min=0"`
+
+	// Disk flush throughput limit in Mb/s
+	ThroughputLimitMbps float64 `yaml:"throughputLimitMbps" validate:"min=0.0"`
+
+	// Disk flush throughput check interval
+	ThroughputCheckEvery int `yaml:"throughputCheckEvery" validate:"nonzero"`
+
+	// NewFileMode is the new file permissions mode to use when
+	// creating files - be sure to specifying leading zero, e.g. 0666.
```

Comment:
the comment says to include "0" in the value, but you're checking length = 3 and prepending "0" whilst parsing. Could you change one or other

Reply:
Yeah I'lll remove this comment.